src/OrbitBase/GetProcessIdsLinux.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <filesystem>
+#include <optional>
+#include <system_error>
+
+#include "OrbitBase/GetProcessIds.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Result.h"
+#include "absl/strings/numbers.h"
+
+namespace orbit_base {
+
+namespace fs = std::filesystem;
+
+static std::optional<pid_t> ProcEntryToPid(const std::filesystem::directory_entry& entry) {
+  if (!entry.is_directory()) {
+    return std::nullopt;
+  }
+
+  int potential_pid;
+  if (!absl::SimpleAtoi(entry.path().filename().string(), &potential_pid)) {
+    return std::nullopt;
+  }
+
+  if (potential_pid <= 0) {
+    return std::nullopt;
+  }
+
+  return potential_pid;
+}
+
+std::vector<pid_t> GetAllPids() {
+  fs::directory_iterator proc{"/proc"};
+  std::vector<pid_t> pids;
+
+  for (const auto& entry : proc) {
+    auto pid = ProcEntryToPid(entry);
+    if (pid.has_value()) {
```

Comment:
Yeah actually here I meant that in other reviews we had gone for how it was before. Something for another religious discussion in a standup.

Reply:
Oh sorry, I skipped over that when addressing comments. 

I think this looks way more readable as it is now. If scope is an issue one can always add additional {}. Yeah, let's (not) discuss at some point :)